### PR TITLE
Add GitHub workflow and LuaCheck

### PR DESCRIPTION
**Pull request by Panquesito7**
Things added/changed:

- Add GitHub workflow and [LuaCheck](https://github.com/luarocks/luacheck).
- Fix warnings reported by [LuaCheck](https://github.com/luarocks/luacheck).

I'm still not so sure how to fix this [error](https://github.com/Panquesito7/staff/runs/1013882965#step:4:7) reported by [LuaCheck](https://github.com/luarocks/luacheck).
@fluxionary please review this as well. Thanks. 🙂

### CODE REVIEW

File: .luacheckrc [hunk 1]
```diff
@@ -0,0 +1,14 @@
+unused_args = false
+allow_defined_top = true
+max_line_length = 999
+
+globals = {
+    "minetest",
+}
+
+read_globals = {
+	string = {fields = {"split", "trim"}},
```
Comment (luk3yx) [inline .luacheckrc]: Mixed spaces+tabs

**Review by luk3yx — CHANGES_REQUESTED** [review]

File: .luacheckrc (hunk 1 above)
Comment (Panquesito7) [reply .luacheckrc]: Fixed, sorry for this always happening. 😄

**Comment by luk3yx** [comment]
Thanks

**Comment by luk3yx** [comment]
I'm just going to make luacheck ignore that loop for now.

**Pull request merged**